libs/core/langchain_core/prompts/chat.py

```diff
@@ -231,7 +231,7 @@ def input_variables(self) -> List[str]:
         Returns:
             List of input variable names.
         """
-        return [self.variable_name] if not self.optional else []
+        return [self.variable_name]
```

Comment:
Here it looks like `input_variables` represents all possible values into the template rather than only required values (which is opposite: 0.2.11  to the convention above).

Reply:
Yeap this is fixed. I see your point